crypto/x509/v3_authattid.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 1999-2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/asn1t.h>
+#include <openssl/x509v3.h>
+#include <crypto/x509_acert.h>
+#include <openssl/x509_acert.h>
+#include "crypto/asn1.h"
+#include "ext_dat.h"
+
+DECLARE_ASN1_ITEM(OSSL_ISSUER_SERIAL)
+
+ASN1_ITEM_TEMPLATE(OSSL_AUTHORITY_ATTRIBUTE_ID_SYNTAX) =
+    ASN1_EX_TEMPLATE_TYPE(ASN1_TFLG_SEQUENCE_OF, 0, OSSL_AUTHORITY_ATTRIBUTE_ID_SYNTAX, OSSL_ISSUER_SERIAL)
+ASN1_ITEM_TEMPLATE_END(OSSL_AUTHORITY_ATTRIBUTE_ID_SYNTAX)
+
+IMPLEMENT_ASN1_FUNCTIONS(OSSL_AUTHORITY_ATTRIBUTE_ID_SYNTAX)
+
+static int i2r_ISSUER_SERIAL(X509V3_EXT_METHOD *method,
+                             OSSL_ISSUER_SERIAL *iss,
+                             BIO *out, int indent)
+{
+    if (iss->issuer != NULL) {
+        BIO_printf(out, "%*sIssuer Names:\n", indent, "");
```

Comment:
Is it intentional not to print `Issuer Names:` header if `->issuer` is NULL. The reason I'm asking we continue to print all other portions (`Issuer Serial`), ...

I think printing either nothing, like function would bail out if `->issuer == NULL` or just empty header would be more friendly to scripts which parse the output. My expectation is the function will either
  - either produce all headers (`Issuer Names:`, `Issuer Serial:` and `Issuer UID:`), and it's fine for me if some values will be missing.
  - or function will produce nothing when `->issuer` is NULL.

Not sure if things I'm asking are reasonable, comply with standard/or existing code.
thank you for considering my suggestion.

Reply:
Between the two, I would prefer to display all headers. Would it be bad form for me to print something like `<empty>`?

Reply:
From my point of view printing nothing when `->issuer` is NULL is perfectly fine. It's also perhaps easier for script authors to detect if `issuer` attribute is present in the certificate.

Printing `<empty>` (or something similar) is also fine. The only thing that needs to be considered is what is the valid value for `issuer`. For example how one can tell the difference from case when issuer is not present from case when someone deliberately selects issuer name '<empty>' I agree it's example but it just illustrates why printing nothing when there is no issuer attribute is better.

but let's wait for other people what do they think.

Reply:
One more thought/reason why I prefer to print nothing when the issuer field is not present: once this change arrives, the output (as I understand) will change for certificates with no issuer too. Not sure if it is desired outcome. Therefore I prefer to print nothing if no issuer is set.

Reply:
I would be OK with printing `<none>` here. I do not think just immediately returning here is correct.

Reply:
Okay, fixed. I also fixed the same code in `v3_ac_tgt.c`.